client/extensions/woocommerce/app/store-stats/index.js

```diff
@@ -98,30 +102,44 @@ class StoreStats extends Component {
 										.format( 'YYYY-MM-DD' )
 								: selectedDate
 						}
-						query={ ordersQuery }
+						query={ query }
 						statsType="statsOrders"
 						showQueryDate
 					/>
 				</StatsPeriodNavigation>
 				<div className="store-stats__widgets">
-					{ sparkWidgets.map( ( widget, index ) => (
-						<div className="store-stats__widgets-column spark-widgets" key={ index }>
-							<Module
+					<div className="store-stats__widgets-column widgets">
+						<Module
+							siteId={ siteId }
+							emptyMessage={ translate( 'No data found' ) }
+							query={ query }
+							statType="statsOrders"
+						>
+							<WidgetList
 								siteId={ siteId }
-								emptyMessage={ translate( 'No data found' ) }
-								query={ ordersQuery }
+								query={ query }
+								selectedDate={ endSelectedDate }
 								statType="statsOrders"
-							>
-								<WidgetList
-									siteId={ siteId }
-									query={ ordersQuery }
-									selectedDate={ endSelectedDate }
-									statType="statsOrders"
-									widgets={ widget }
-								/>
-							</Module>
-						</div>
-					) ) }
+								widgets={ sparkWidgets }
+							/>
+						</Module>
+					</div>
+					<div className="store-stats__widgets-column widgets">
+						<Module
+							siteId={ siteId }
+							emptyMessage={ translate( 'No data found' ) }
+							query={ query }
+							statType="statsStoreReferrers"
+							header={ referrerHeader }
+						>
+							<StoreStatsReferrerWidget
```

Comment:
Should we have a feature flag around this while we are still working on it, and the API bits?

Reply:
I was avoiding it because the widget changes the layout, but I think now its a good idea. I can reapply layout changes as the very last step